src/control/Control.Search.js

```diff
@@ -47,28 +47,42 @@ export var Search = Control.extend({
       region: 'left'
     });
 
+    this._reader.on('update-section', function() {
+      self._reader.annotations.reset = function() {
+        for(var hash in reader.annotations._annotations) {
+            var cfiRange = decodeURI(hash);
+            reader.annotations.remove(cfiRange);
```

Comment:
I think this works because `reader` is in the global context. If you used `self._reader` that should be safe.